cmake/reports/CMakeLists.txt

```diff
@@ -12,6 +12,7 @@ foreach(report ram_report rom_report)
     -k ${ZEPHYR_BINARY_DIR}/${KERNEL_ELF_NAME}
     -z ${ZEPHYR_BASE}
     -o ${CMAKE_BINARY_DIR}
+    $<$<BOOL:${WEST_TOPDIR}>:--westtopdir=${WEST_TOPDIR}>
```

Comment:
Should this be a made a bit more generic ?

size_report itself doesn't have any dependencies on west, and it could probably be useful for users not using west to still specify a workspace dir, as example using `-DZEPHYR_WORKSPACE=<workspace path>`.

Also, we already at this stage know whether or not WEST_TOPDIR is defined, so there is little value in using a generator expression here.
Generator expressions are very powerful when it comes to avoid configure time process order dependencies, as example allowing target properties to be set late in cmake.
But they are harder to read for most users.

So I propose to change into:
```
...
set(report_depth 99)
if(DEFINED ZEPHYR_WORKSPACE)
  set(workspace_arg "--workspace=${ZEPHYR_WORKSPACE}")
elseif(DEFINED WEST_TOPDIR)
  set(workspace_arg "--workspace=${WEST_TOPDIR}")
endif()

foreach(report ram_report rom_report)
  add_custom_target(
    ${report}
    ${PYTHON_EXECUTABLE}
    ${ZEPHYR_BASE}/scripts/footprint/size_report
    -k ${ZEPHYR_BINARY_DIR}/${KERNEL_ELF_NAME}
    -z ${ZEPHYR_BASE}
    -o ${CMAKE_BINARY_DIR}
    ${workspace_arg}
   ....
```
which also follows the pattern for other arguments, such as `flag_for_${report}` and `report_depth`.